app/api/v2/players.py

```diff
@@ -1,11 +1,52 @@
 """ api-v2: redefined bancho.py's developer api for interacting with players repository """
 from __future__ import annotations
 
+from typing import List
+from typing import Optional
+from typing import Union
+
 from fastapi import APIRouter
+from fastapi.param_functions import Query
+
+from app.api.v2.models.players import Player
+from app.api.v2.models.players import Players
+from app.api.v2.responses.error import failure
+from app.api.v2.responses.success import Success
+from app.api.v2.responses.success import success
+from app.repositories import players as players_repo
 
 router = APIRouter()
 
 
-@router.get("/players/{player_id}")
-async def get_player(player_id: int):
-    return f"Hello! {player_id}"
+@router.get("/players", response_model=Success[Players])
+async def get_players(
+    page: Optional[int] = Query(1, ge=1),
+    page_size: Optional[int] = Query(50, ge=50, le=100),
+):
+    players = await players_repo.fetch_many(page=page, page_size=page_size)
+
+    if len(players) < 1:
+        return failure("Reached the pages limit.")
+
+    players_count = await players_repo.fetch_count()
+
+    response = Players(
+        players=[Player.from_mapping(x) for x in players],
+        max_pages=players_count / page_size + 1,
+        current_page=page,
+    )
+    return success(response)
+
+
+@router.get("/players/{player}", response_model=Success[Player])
```

Comment:
this is *very* new, but as of fastapi 0.89.0, `response_model` can be replaced with a return type annotation
https://github.com/tiangolo/fastapi/releases/tag/0.89.0